reconcile/status_page_components.py

```diff
@@ -0,0 +1,261 @@
+from abc import abstractmethod
+import logging
+from typing import Optional, Any
+from collections.abc import Iterable
+import sys
+
+from pydantic import BaseModel, Field
+from pydantic.networks import HttpUrl
+import statuspageio  # type: ignore
+
+from reconcile import queries
+from reconcile.utils.semver_helper import make_semver
+from reconcile.utils.state import State
+from reconcile.utils.vaultsecretref import VaultSecretRef
+
+
+QONTRACT_INTEGRATION = "status-page-components"
+QONTRACT_INTEGRATION_VERSION = make_semver(0, 1, 0)
+
+
+LOG = logging.getLogger(__name__)
+
+
+class AtlassianComponent(BaseModel):
+    """
+    atlassian status page REST schema for component
+    """
+
+    id: str
+    name: str
+    description: Optional[str]
+    position: int
+    status: str
+    automation_email: Optional[str]
+    group_id: Optional[str]
+    group: Optional[bool]
+    group_name: Optional[str]
+
+
+class StatusComponent(BaseModel):
+    """
+    app-interface schema dependencies/status-page-component-1
+    """
+
+    name: str
+    display_name: str = Field(..., alias="displayName")
+    description: Optional[str]
+    group_name: Optional[str] = Field(..., alias="groupName")
+    component_id: Optional[str]
+
+
+class StatusPageProvider(BaseModel):
+    """
+    Provider specific status page reconcile implementation is implemented
+    as a subclass of `StatusPageProvider`
+    """
+
+    @abstractmethod
+    def component_ids(self) -> Iterable[str]:
+        return []
+
+    @abstractmethod
+    def apply_component(self, dry_run: bool,
+                        desired: StatusComponent) -> Optional[str]:
+        return None
+
+    @abstractmethod
+    def delete_component(self, dry_run: bool, id: str):
```

Comment:
Make the return type explicit here:
```suggestion
    def delete_component(self, dry_run: bool, id: str) -> None:
```

Reply:
fixed